### PR TITLE
openvpn tunnels

### PR DESCRIPTION
**Pull request by cotosso**
OpoenVPN tunnel: first manual page after UI release.

### CODE REVIEW

File: openvpn_tunnels.rst [hunk 1]
```diff
@@ -1,12 +1,69 @@
...
 OpenVPN net-to-net tunnels establish secure connections between two separate networks, such as branch offices of a company, over the internet.
 These connections use SSL/TLS protocols for encryption and authentication, ensuring data confidentiality and integrity.
 
-See the `developer manual <https://dev.nethsecurity.org/packages/ns-openvpn/#openvpn-tunnels>`_ for more info on manual configuration.
+The connection is handled by 2 NethSecurity firewalls, each one has a specific role.
+When creating an OpenVPN net2net connection a firewall will have the master role (server) whereas the other NethSecurity will connect to it as a client.
+One NethSecurity can be the same time server and client for different tunnels, all tunnels use OpenVPN routed mode.
+
+
+.. note:: On NethSecurity the OpenVPN tunnels UI has been designed to make connecting 2 Nethesis devices very simple. For this reason, it is deliberately limited and does not expose all the parameters that can be configured with OpenVPN to connect to any third-party device.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:15]: The sentence is ok, but I'id remove the note.
```suggestion
On NethSecurity the OpenVPN tunnels UI has been designed to make connecting 2 Nethesis devices very simple. For this reason, it is deliberately limited and does not expose all the parameters that can be configured with OpenVPN to connect to any third-party device.
```

File: openvpn_tunnels.rst [hunk 2]
```diff
@@ -1,12 +1,69 @@
...
+To connect to a third-party device, it's recommended the use of IPsec protocol.
+
+Configuration
+-------------
+To connect 2 firewalls via an openvpn tunnel, first configure the server firewall, then configure the client firewall.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:20]:
```suggestion
To connect 2 firewalls via an OpenVPN tunnel, first configure the server firewall, then configure the client one.
```

File: openvpn_tunnels.rst [hunk 3]
```diff
@@ -1,12 +1,69 @@
...
+The server must have at least one public IP to be contacted by the client, while the client may not even have public IPs.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:21]:
```suggestion
The server needs at least one public IP address to be accessible to the client, while the client may not even have public IPs.
```

File: openvpn_tunnels.rst [hunk 4]
```diff
@@ -1,12 +1,69 @@
...
+The configuration of the firewall server requires only a very few parameters, where possible all the parameters are already filled in automatically to avoid errors and speed up the process.
+Once the server firewall has been configured, it will be possible to download the client configuration to import onto the other firewall.
+
+Proceed as follows:
+^^^^^^^^^^^^^^^^^^^
+Access the OpenVPN tunnels page, move to Tunnel servers tab and click on :guilabel:`Add server tunnel`.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:27]:
```suggestion
Access the OpenVPN tunnels page, move to ``Server tunnel`` tab and click on :guilabel:`Add server tunnel`.
```

File: openvpn_tunnels.rst [hunk 5]
```diff
@@ -1,12 +1,69 @@
...
+
+Insert all required fields, but please note:
+
+* Public endpoints are a list of IP addresses or hostnames that clients can use to reach the OpenVPN tunnel server
+* Local networks it’s a list of local networks that will be accessible from the remote server. If topology is set to p2p, the same list will be reported inside the client Remote networks field
+* Remote networks, it’s a list of networks behind the remote server which will be accessible from hosts in the local network
+* After the configuration is saved, click on the :guilabel:`Download` action and select Client configuration
+* Access the client firewall, OpenVPN tunnel,  move to Client tunnel tab, click on :guilabel:`Import configuration`
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:35]:
```suggestion
* ``Public endpoints`` it's a list of IP addresses or hostnames that clients can use to reach the OpenVPN tunnel server
* ``Local networks`` it's a list of local networks that will be accessible from the remote server. If topology is set to p2p, the same list will be reported inside the client ``Remote networks`` field
* ``Remote networks``, it’s a list of networks behind the remote server which will be accessible from hosts in the local network
* After the configuration is saved, click on the :guilabel:`Download` action and select ``Client configuration``
* Access the client firewall, OpenVPN tunnel,  move to ``Client tunnel`` tab, click on :guilabel:`Import configuration`
```

File: openvpn_tunnels.rst [hunk 6]
```diff
@@ -1,12 +1,69 @@
...
+
+Topology
+--------
+Tunnels can have two kinds of topologies: subnet and p2p (Point to Point).
+
+Subnet
+^^^^^^
+Subnet is the default topology and the recommended one: in subnet topology, the server will accept connections and will act as a DHCP server for every connected clients.
+
+In this scenario the server will authenticate clients using TLS certificates and will push local routes to remote clients.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:45]:
```suggestion
In this scenario the server will authenticate clients using TLS certificates and will push local routes to remote client.
```

File: openvpn_tunnels.rst [hunk 7]
```diff
@@ -1,12 +1,69 @@
...
+The client will be able to authenticate with TLS certificates or user name and password.
+
+P2P
+^^^
+
+In a p2p topology, the administrator must configure one server for each client, in this scenario the only supported authentication method is the PSK (Pre-Shared Key).
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:51]:
```suggestion
In a ``p2p`` topology, the administrator must configure one server for each client, in this scenario the only supported authentication method is the PSK (Pre-Shared Key). 
```

File: openvpn_tunnels.rst [hunk 8]
```diff
@@ -1,12 +1,69 @@
...
+Subnet is the default topology and the recommended one: in subnet topology, the server will accept connections and will act as a DHCP server for every connected clients.
+
+In this scenario the server will authenticate clients using TLS certificates and will push local routes to remote clients.
+The client will be able to authenticate with TLS certificates or user name and password.
+
+P2P
+^^^
+
+In a p2p topology, the administrator must configure one server for each client, in this scenario the only supported authentication method is the PSK (Pre-Shared Key). 
+Please make sure to exchange the PSK using a secure channel (like SSH or HTTPS) the administrator must select an IP for both end points routes to remote networks must be configured on each end point.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:52]: This is not clear: the sentence is mixing 2 different concepts.

File: openvpn_tunnels.rst [hunk 9]
```diff
@@ -1,12 +1,69 @@
...
+Topology
+--------
+Tunnels can have two kinds of topologies: subnet and p2p (Point to Point).
+
+Subnet
+^^^^^^
+Subnet is the default topology and the recommended one: in subnet topology, the server will accept connections and will act as a DHCP server for every connected clients.
+
+In this scenario the server will authenticate clients using TLS certificates and will push local routes to remote clients.
+The client will be able to authenticate with TLS certificates or user name and password.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:46]: AFAIK this is not true: tunnels do not support name and password auth.

File: openvpn_tunnels.rst [hunk 10]
```diff
@@ -1,12 +1,69 @@
...
+
+In a p2p topology, the administrator must configure one server for each client, in this scenario the only supported authentication method is the PSK (Pre-Shared Key). 
+Please make sure to exchange the PSK using a secure channel (like SSH or HTTPS) the administrator must select an IP for both end points routes to remote networks must be configured on each end point.
+
+
+Advanced features
+-----------------
+The web interface allows the configuration of advanced features like:
+
+* Multiple remote host: Multiple remote server addresses can be specified for redundancy; the OpenVPN client will try to connect to each host in the given order
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:59]:
```suggestion
* ``Multiple remote host``: multiple remote server addresses can be specified for redundancy; the OpenVPN client will try to connect to each host in the given order
```

File: openvpn_tunnels.rst [hunk 11]
```diff
@@ -1,12 +1,69 @@
...
+
+* Protocol: OpenVPN is designed to operate optimally over UDP, but TCP capability is provided for situations where UDP cannot be used
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:61]:
```suggestion
* ``Protocol``: OpenVPN is designed to operate optimally over UDP, but TCP capability is provided for situations where UDP cannot be used
```

File: openvpn_tunnels.rst [hunk 12]
```diff
@@ -1,12 +1,69 @@
...
+
+* Cipher: the cryptographic algorithm used to encrypt all the traffic. If not explicitly selected, the server and client will try to negotiate the best cipher available on both sides
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:63]:
```suggestion
* ``Cipher``: the cryptographic algorithm used to encrypt all the traffic. If not explicitly selected, the server and client will try to negotiate the best cipher available on both sides
```

File: openvpn_tunnels.rst [hunk 13]
```diff
@@ -1,12 +1,69 @@
...
+
+
+
+
+
+
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:69]: Remove empty lines and add missing fields:
- compression
- digest
- tls version

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: openvpn_tunnels.rst (hunk 9 above)
Comment (cotosso) [reply openvpn_tunnels.rst:46]: the client tunnel in the UI shows both methods: "certificate" and "username, password and certificate"

Comment (gsanchietti) [reply openvpn_tunnels.rst:46]: I didn't remember that, I think it's not going to work

File: openvpn_tunnels.rst (hunk 13 above)
Comment (cotosso) [reply openvpn_tunnels.rst]: Fixed

File: openvpn_tunnels.rst (hunk 9 above)
Comment (cotosso) [reply openvpn_tunnels.rst]: Last line has been removed

File: openvpn_tunnels.rst [hunk 14]
```diff
@@ -1,12 +1,71 @@
...
 OpenVPN net-to-net tunnels establish secure connections between two separate networks, such as branch offices of a company, over the internet.
 These connections use SSL/TLS protocols for encryption and authentication, ensuring data confidentiality and integrity.
 
-See the `developer manual <https://dev.nethsecurity.org/packages/ns-openvpn/#openvpn-tunnels>`_ for more info on manual configuration.
+The connection is handled by 2 NethSecurity firewalls, each one has a specific role.
+When creating an OpenVPN net2net connection a firewall will have the master role (server) whereas the other NethSecurity will connect to it as a client.
+One NethSecurity can be the same time server and client for different tunnels, all tunnels use OpenVPN routed mode.
+
+
+On NethSecurity the OpenVPN tunnels UI has been designed to make connecting 2 Nethesis devices very simple. For this reason, it is deliberately limited and does not expose all the parameters that can be configured with OpenVPN to connect to any third-party device.
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:15]:
```suggestion
The OpenVPN tunnels interface has been crafted for straightforward connection between two NethSecuirty devices. For this reason, it is deliberately limited and does not expose all the parameters that can be configured with OpenVPN to connect to any third-party device.
```

File: openvpn_tunnels.rst [hunk 15]
```diff
@@ -1,12 +1,71 @@
...
+-----------------
+The web interface allows the configuration of advanced features like:
+
+* ``Multiple remote host``: multiple remote server addresses can be specified for redundancy; the OpenVPN client will try to connect to each host in the given order
+
+* ``Protocol``: OpenVPN is designed to operate optimally over UDP, but TCP capability is provided for situations where UDP cannot be used
+
+* ``Compression``: if enabled, data to be sent through the VPN tunnel will be compressed. This option is disabled by default both for security reasons and because it is almost never necessary (today internet traffic is already highly compressed and optimized).
+
+* ``Digest``: the digest algorithm used to transform an arbitrarily large block of data into a fixed-size output.If not explicitly selected, the server and client will try to negotiate the best cipher available on both sides
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:67]:
```suggestion
* ``Digest``: the digest algorithm used to transform an arbitrarily large block of data into a fixed-size output. If not explicitly selected, the server and client will try to negotiate the best digest available on both sides
```

File: openvpn_tunnels.rst [hunk 16]
```diff
@@ -1,12 +1,71 @@
...
+
+Advanced features
+-----------------
+The web interface allows the configuration of advanced features like:
+
+* ``Multiple remote host``: multiple remote server addresses can be specified for redundancy; the OpenVPN client will try to connect to each host in the given order
+
+* ``Protocol``: OpenVPN is designed to operate optimally over UDP, but TCP capability is provided for situations where UDP cannot be used
+
+* ``Compression``: if enabled, data to be sent through the VPN tunnel will be compressed. This option is disabled by default both for security reasons and because it is almost never necessary (today internet traffic is already highly compressed and optimized).
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:65]:
```suggestion
* ``Compression``: if enabled, data to be sent through the VPN tunnel will be compressed. This option is disabled by default both for security reasons. Compression is rarely essential nowadays, as internet traffic is typically already highly compressed and optimized.
```

File: openvpn_tunnels.rst [hunk 17]
```diff
@@ -1,12 +1,71 @@
...
+
+* ``Protocol``: OpenVPN is designed to operate optimally over UDP, but TCP capability is provided for situations where UDP cannot be used
+
+* ``Compression``: if enabled, data to be sent through the VPN tunnel will be compressed. This option is disabled by default both for security reasons. Compression is rarely essential nowadays, as internet traffic is typically already highly compressed and optimized.
+
+* ``Digest``: the digest algorithm used to transform an arbitrarily large block of data into a fixed-size output. If not explicitly selected, the server and client will try to negotiate the best digest available on both sides
+
+* ``Cipher``: the cryptographic algorithm used to encrypt all the traffic. If not explicitly selected, the server and client will try to negotiate the best cipher available on both sides
+
+* ``Enforce a minimum TLS version``: Allows you to choose a minimum version of TLS, in which case connections will only be allowed from devices that use a version greater than or equal to the one reported
```
Comment (gsanchietti) [inline openvpn_tunnels.rst:71]:
```suggestion
* ``Enforce a minimum TLS version``: Allows you to choose a minimum version of TLS, in which case connections will only be allowed from devices that use a version greater than or equal to the one selected
```

**Review by gsanchietti — APPROVED** [review]

File: openvpn_tunnels.rst [hunk 18]
```diff
@@ -1,12 +1,71 @@
...
 These connections use SSL/TLS protocols for encryption and authentication, ensuring data confidentiality and integrity.
 
-See the `developer manual <https://dev.nethsecurity.org/packages/ns-openvpn/#openvpn-tunnels>`_ for more info on manual configuration.
+The connection is handled by 2 NethSecurity firewalls, each one has a specific role.
+When creating an OpenVPN net2net connection a firewall will have the master role (server) whereas the other NethSecurity will connect to it as a client.
+One NethSecurity can be the same time server and client for different tunnels, all tunnels use OpenVPN routed mode.
+
+
+The OpenVPN tunnels interface has been crafted for straightforward connection between two NethSecuirty devices. For this reason, it is deliberately limited and does not expose all the parameters that can be configured with OpenVPN to connect to any third-party device.
+To connect to a third-party device, it's recommended the use of IPsec protocol.
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:16]:
```suggestion
To connect to a third-party device, it's recommended to use the IPsec protocol.
```

File: openvpn_tunnels.rst [hunk 19]
```diff
@@ -1,12 +1,71 @@
...
+
+Configuration
+-------------
+To connect 2 firewalls via an OpenVPN tunnel, first configure the server firewall, then configure the client one.
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:20]:
```suggestion
To connect two firewalls via an OpenVPN tunnel, first configure the server firewall, then configure the client one.
```

File: openvpn_tunnels.rst [hunk 20]
```diff
@@ -1,12 +1,71 @@
...
-.. warning::
-
-   This feature is still under development and does not have a user interface yet. It can currently only be configured through the command line.
-
 OpenVPN net-to-net tunnels establish secure connections between two separate networks, such as branch offices of a company, over the internet.
 These connections use SSL/TLS protocols for encryption and authentication, ensuring data confidentiality and integrity.
 
-See the `developer manual <https://dev.nethsecurity.org/packages/ns-openvpn/#openvpn-tunnels>`_ for more info on manual configuration.
+The connection is handled by 2 NethSecurity firewalls, each one has a specific role.
+When creating an OpenVPN net2net connection a firewall will have the master role (server) whereas the other NethSecurity will connect to it as a client.
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:11]:
```suggestion
When creating an OpenVPN net2net connection a firewall will have the server role whereas the other NethSecurity will connect to it as a client.
```

File: openvpn_tunnels.rst [hunk 21]
```diff
@@ -1,12 +1,71 @@
...
+One NethSecurity can be the same time server and client for different tunnels, all tunnels use OpenVPN routed mode.
+
+
+The OpenVPN tunnels interface has been crafted for straightforward connection between two NethSecuirty devices. For this reason, it is deliberately limited and does not expose all the parameters that can be configured with OpenVPN to connect to any third-party device.
+To connect to a third-party device, it's recommended the use of IPsec protocol.
+
+Configuration
+-------------
+To connect 2 firewalls via an OpenVPN tunnel, first configure the server firewall, then configure the client one.
+The server needs at least one public IP address to be accessible to the client, while the client may not even have public IPs.
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:21]:
```suggestion
The server needs at least one public IP address to be reachable by the client, while the client may not even have public IPs.
```

File: openvpn_tunnels.rst [hunk 22]
```diff
@@ -1,12 +1,71 @@
...
+^^^^^^^^^^^^^^^^^^^
+Access the OpenVPN tunnels page, move to ``Server tunnel`` tab and click on :guilabel:`Add server tunnel`.
+
+Insert all required fields, but please note:
+
+* ``Public endpoints`` it's a list of IP addresses or hostnames that clients can use to reach the OpenVPN tunnel server
+* ``Local networks`` it's a list of local networks that will be accessible from the remote server. If topology is set to p2p, the same list will be reported inside the client ``Remote networks`` field
+* ``Remote networks``, it’s a list of networks behind the remote server which will be accessible from hosts in the local network
+* After the configuration is saved, click on the :guilabel:`Download` action and select ``Client configuration``
+* Access the client firewall, OpenVPN tunnel,  move to ``Client tunnel`` tab, click on :guilabel:`Import configuration`
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:35]:
```suggestion
* Access the client firewall, OpenVPN tunnel, move to ``Client tunnel`` tab, click on :guilabel:`Import configuration`
```

File: openvpn_tunnels.rst [hunk 23]
```diff
@@ -1,12 +1,71 @@
...
+
+Topology
+--------
+Tunnels can have two kinds of topologies: ``subnet`` and ``p2p`` (Point to Point).
+
+Subnet
+^^^^^^
+``Subnet`` is the default topology and the recommended one: in ``subnet`` topology, the server will accept connections and will act as a DHCP server for every connected clients.
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:43]:
```suggestion
``Subnet`` is the default topology and the recommended one: in ``subnet`` topology, the server will accept connections and will act as a DHCP server for every connected client.
```

File: openvpn_tunnels.rst [hunk 24]
```diff
@@ -1,12 +1,71 @@
...
+In this scenario the server will authenticate clients using TLS certificates and will push local routes to remote client.
+
+P2P
+^^^
+
+In a ``p2p`` topology, the administrator must configure one server for each client, in this scenario the only supported authentication method is the PSK (Pre-Shared Key). 
+
+* make sure to exchange the PSK using a secure channel (like SSH or HTTPS) 
+* the administrator must select an IP for both end points 
+* routes to remote networks must be configured on each end point
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:54]:
```suggestion
* routes to remote networks must be configured on each endpoint
```

File: openvpn_tunnels.rst [hunk 25]
```diff
@@ -1,12 +1,71 @@
...
+
+In this scenario the server will authenticate clients using TLS certificates and will push local routes to remote client.
+
+P2P
+^^^
+
+In a ``p2p`` topology, the administrator must configure one server for each client, in this scenario the only supported authentication method is the PSK (Pre-Shared Key). 
+
+* make sure to exchange the PSK using a secure channel (like SSH or HTTPS) 
+* the administrator must select an IP for both end points
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:53]:
```suggestion
* the administrator must select an IP for both endpoints 
```

File: openvpn_tunnels.rst [hunk 26]
```diff
@@ -1,12 +1,71 @@
...
+
+Advanced features
+-----------------
+The web interface allows the configuration of advanced features like:
+
+* ``Multiple remote host``: multiple remote server addresses can be specified for redundancy; the OpenVPN client will try to connect to each host in the given order
+
+* ``Protocol``: OpenVPN is designed to operate optimally over UDP, but TCP capability is provided for situations where UDP cannot be used
+
+* ``Compression``: if enabled, data to be sent through the VPN tunnel will be compressed. This option is disabled by default both for security reasons. Compression is rarely essential nowadays, as internet traffic is typically already highly compressed and optimized.
```
Comment (filippocarletti) [inline openvpn_tunnels.rst:65]:
```suggestion
* ``Compression``: if enabled, data to be sent through the VPN tunnel will be compressed. This option is disabled by default both for security reasons. Compression is rarely essential nowadays, as internet traffic is typically already highly compressed and optimized
```

**Review by filippocarletti — CHANGES_REQUESTED** [review]

**Review by filippocarletti — APPROVED** [review]

**Pull request merged**